src/profiles.js

```diff
@@ -0,0 +1,195 @@
+import tv4 from 'tv4'
+import _ from 'lodash'
+import path from 'path'
+
+import Utils from './utils'
+
+const PROFILES_CACHED = {}
+  , DEFAULT_REMOTE_PATH = 'http://schemas.datapackages.org/registry.csv'
+let DEFAULT_LOCAL_PATH
+if (!Utils.isBrowser) {
+  DEFAULT_LOCAL_PATH = path.join(__dirname, '..', 'schemas', 'registry.csv')
+}
+
+/**
+ * Base class for retrieving profile schemas and validating datapackage
+ * descriptors against profiles.
+ *
+ * @returns {Promise}
+ */
+class Profiles {
+  constructor(remote = false) {
+    const self = this
+        , PATH = (remote || Utils.isBrowser) ? DEFAULT_REMOTE_PATH :
+                 DEFAULT_LOCAL_PATH
+
+    return new Promise((resolve, reject) => {
+      this._loadRegistry(PATH).then(registry => {
+        self._registry = registry
+        self._basePath = Utils.getDirname(PATH)
+        this._getProfiles().then(profiles => {
+          self._allProfiles = profiles
+          resolve(self)
+        }).catch(err => {
+          reject(err)
+        })
+      }).catch(err => {
+        reject(err)
+      })
+    })
+  }
+
+  /**
+   * Retrieve a profile by id.
+   *
+   * @param {String} [profile='base']
+   * @return {Object}
+   */
+  retrieve(profile = 'base') {
+    return this._allProfiles[profile] || null
+  }
+
+  /**
+   * Validate a descriptor against a profile. You can provide custom schema
+   * or specify the profile id.
+   *
+   * @param {Object} descriptor The descriptor that needs to be validated
+   * @param {Object} profile Schema to validate against
```

Comment:
> We don't need to support descriptor as a string only object. Also you use JSON to say object is expected. Not sure it's correct (not sure of otherwise).

Descriptor: Object but here I'm pointing on profile wich 'base' (string) by default and should accept profile names as strings. I suppose it's just a typo in docstring.